### ISSUE TITLE
[EPIC] New Escrow migration contract with L2 support

### ISSUE THREAD

**Issue by jacko125** [Synthetixio/synthetix-pm#144]
- [x] (Ale) Research data migration technique (can we send that much data)
  * TL;DR Looks like we're good to go 👍 
  * Data format: 2 arrays, 52 entries each, `uint64[52] timestamps` and `uint256[52] amounts`
  * Experiment 👉 [escrow-migration-poc]( https://github.com/ajsantander/escrow-migration-poc)  
  * The experiment shows that storing the data via a function with the 2 arrays as parameters spends 3.7m gas in L2
  * This means that ~400k gas would need to be reserved on L1 for the operation
  * Unless we think this is too much gas to spend on L1, we don't need to do anything
- [x] (Yannis) ~Add ability to update hardcoded gas value in the deposit contract~ UPDATE: Tracked in Epic #148
- [ ] (Ale) Study the possibility to compress/pack the payload
- [x] (Jacko) [Working document on the L1 requirements](https://docs.google.com/document/d/1p3BDyvKYNxxvkkh3fMD8r9w6u9WMESfeeU8UEm4Msc0/edit?ts=5f88df00#)
- [x] (Jacko) (MERGED) SIP: https://github.com/Synthetixio/SIPs/pull/270
- [ ] (Jacko) New V2 Escrow contract: https://github.com/Synthetixio/synthetix/compare/sip-60-reward-escrow-v2
  * returns the vesting schedules to deposit bridge to L2
  * supports the removal of entries for migration to L2
- [ ] (Yannis) Integrate V2 Escrow into L1 and L2 https://github.com/Synthetixio/synthetix/tree/test-escrow-migration
